service/start.go

```diff
@@ -113,6 +146,73 @@ func (svc *service) startNostr(ctx context.Context) error {
 	return nil
 }
 
+func (svc *service) startAllExistingAppsWalletSubscriptions(ctx context.Context, relay *nostr.Relay) {
+	var apps []db.App
+	result := svc.db.Where("wallet_pubkey IS NOT NULL").Find(&apps)
+	if result.Error != nil {
+		logger.Logger.WithError(result.Error).Error("Failed to fetch App records with non-empty WalletPubkey")
+		return
+	}
+
+	for _, app := range apps {
+		go func(app db.App) {
+			err := svc.startAppWalletSubscription(ctx, relay, *app.WalletPubkey)
+			if err != nil {
+				logger.Logger.WithError(err).WithFields(logrus.Fields{
+					"app_id": app.ID}).Error("Subscription error")
+				return
+			}
+		}(app)
+	}
+}
+
+func (svc *service) startAppWalletSubscription(ctx context.Context, relay *nostr.Relay, appWalletPubKey string) error {
+
+	logger.Logger.Info("Subscribing to events for wallet ", appWalletPubKey)
+	sub, err := relay.Subscribe(ctx, svc.createFilters(appWalletPubKey))
+	if err != nil {
+		logger.Logger.WithError(err).Error("Failed to subscribe to events")
+		return err
+	}
+
+	// register a subscriber for "app_deleted" events, which handles nostr subscription cancel and nip47 info event deletion
+	deleteEventSubscriber := deleteAppConsumer{nostrSubscription: sub, walletPubkey: appWalletPubKey, svc: svc, relay: relay}
+	svc.eventPublisher.RegisterSubscriber(&deleteEventSubscriber)
+
+	err = svc.StartSubscription(sub.Context, sub)
+	svc.eventPublisher.RemoveSubscriber(&deleteEventSubscriber)
+	if err != nil {
+		logger.Logger.WithError(err).Error("Got an error from the relay while listening to subscription.")
+		return err
+	}
+	return nil
+}
+
+func (svc *service) StartSubscription(ctx context.Context, sub *nostr.Subscription) error {
+	svc.nip47Service.StartNotifier(ctx, sub.Relay, svc.lnClient)
+
+	go func() {
+		// block till EOS is received
+		<-sub.EndOfStoredEvents
+		logger.Logger.Debug("Received EOS")
+
+		// loop through incoming events
+		for event := range sub.Events {
+			go svc.nip47Service.HandleEvent(ctx, sub.Relay, event, svc.lnClient)
+		}
+		logger.Logger.Debug("Relay subscription events channel ended")
+	}()
+
+	<-ctx.Done()
+
+	if sub.Relay.ConnectionError != nil {
+		logger.Logger.WithField("connectionError", sub.Relay.ConnectionError).Error("Relay error")
+		return sub.Relay.ConnectionError
+	}
+	logger.Logger.Info("Exiting subscription...")
+	return nil
+}
+
 func (svc *service) StartApp(encryptionKey string) error {
```

Comment:
@im-adithya we can't do that because launching the backend relies on the keys.

What about we change this check to also check if the node was run before, and only then start the app?

```
if appConfig.AutoUnlockPassword != "" {
		svc.StartApp(appConfig.AutoUnlockPassword)
	}
```

Reply:
Nice that's better 🙌